src/app/lib/utilities/preprocessor/rules/blockPositioning.js

```diff
@@ -0,0 +1,49 @@
+import deepGet from '../../../../helpers/json/deepGet';
+import deepClone from '../../../../helpers/json/deepClone';
+
+const insertBlockPositioning = (
+  { model },
+  positionArr = [],
+) => {
+  if (!model || typeof model !== 'object') {
+    return null;
+  }
+
+  const amendedJson = model;
+  let incrementPositionAtLevel = 0;
+  Object.keys(model).forEach(key => {
+    if (key !== 'blocks') {
+      return null;
+    }
+
+    return model[key].map((childObj, index) => {
+      const newPosition = [...positionArr, (incrementPositionAtLevel += 1)];
+      const newChildObj = Object.assign(childObj, { position: newPosition });
+
+      amendedJson[key][index] = newChildObj;
+      return insertBlockPositioning(newChildObj, newPosition);
+    });
+  });
+
+  return amendedJson;
+};
+
+/**
+ * This preprocessor rule inserts a `position` property to each block object,
+ * denoting its ordinal position in an array.
+ * @param {Object} obj
+ */
+const applyBlockPositioning = obj => {
+  const json = deepClone(obj);
+  const content = deepGet(['content'], obj);
+
+  if (content && content.model) {
```

Comment:
```suggestion
  if (deepGet(['model'], content)) {
```

Reply:
For the same reason above, this won't work. Similarly, `'model' in content` will not work either without having a check that `content` is truthy, therefore not much point changing it imo.

Reply:
Hmmm good point, I think we should extend the `deepGet` method to check that the thing passed in as the `obj` param is actually an `object`. I've opened the issue https://github.com/bbc/simorgh/issues/1913